components/homme/src/theta-l_kokkos/cxx/Diagnostics.cpp

```diff
@@ -90,10 +96,13 @@ void Diagnostics::init_buffers (const FunctorsBuffersManager& fbm) {
   }
 }
 
-void Diagnostics::sync_diags_to_host () {
-  Kokkos::deep_copy(h_IEner,m_IEner);
-  Kokkos::deep_copy(h_PEner,m_PEner);
-  Kokkos::deep_copy(h_KEner,m_KEner);
+void Diagnostics::sync_diagnostics_to_host () {
+  Kokkos::deep_copy(h_IEner,  d_IEner);
+  Kokkos::deep_copy(h_PEner,  d_PEner);
+  Kokkos::deep_copy(h_KEner,  d_KEner);
+  Kokkos::deep_copy(h_Qvar,   d_Qvar);
+  Kokkos::deep_copy(h_Qmass,  d_Qmass);
+  Kokkos::deep_copy(h_Q1mass, d_Q1mass);
```

Comment:
Remove all deep copies from the actual computation and add here. This function wasn't previously used, but now it's called right before output to screen.